File: cf/round704/b.cpp
#include <iostream>
using namespace std;

void solve(){
    long long p, a, b, c;
    cin >> p >> a >> b >> c;
    if(p % a == 0 || p % b == 0 || p % c == 0){
        cout << 0 << endl;
        return;
    }
    if(p < a) a = a - p;
    if(p < b) b = b - p;
    if(p < c) c = c - p;
    if(p > a){
        long long x = p / a;
        a = (x + 1) * a - p;
    }
    if(p > b){
        long long x = p / b;
        b = (x + 1) * b - p;
    }
    if(p > c){
        long long x = p / c;
        c = (x + 1) * c - p;
    }
    cout << min(a, min(b, c)) << endl;
}


int main(){
    ios_base::sync_with_stdio();
    cin.tie(0);
    int n;
    cin >> n;
    while(n--) solve();
}

File: cf/round704/b.py
def solve():
    p, a, b, c = map(int, input().split())
    a1, b1, c1 = False, False, False
    if p % a == 0 or p % b == 0 or p % c == 0:
        print(0)
        return
    x, A, B, C = 1, a, b, c
    while 1:
        if A > p and not a1:
            a1 = True 
            A -= p
        if B > p and not b1:
            b1 = True 
            B -= p
        if C > p and not c1:
            c1 = True 
            C -= p
        if a1 and b1 and c1:
            break
        x += 1
        if not a1:
            A = x * a
        if not b1:
            B = x * b
        if not c1:
            C = x * c
    print(min(A, B, C))
        


for i in range(int(input())):
    solve()